cle/backends/elf/elf.py

```diff
@@ -459,68 +466,14 @@ def _load_function_hints_from_fde(self, dwarf):
     # will probably break things. Caveat emptor.
     #
 
-    def __getstate__(self):
-        if self.binary is None:
-            raise ValueError("Can't pickle an object loaded from a stream")
-
-        state = dict(self.__dict__)
-
-        # Trash the unpickleable
-        if type(self.binary_stream) is PatchedStream:
-            state['binary_stream'].stream = None
-        else:
-            state['binary_stream'] = None
-
-        state['reader'] = None
-        state['strtab'] = None
-        state['dynsym'] = None
-        state['hashtable'] = None
-
-        return state
-
-    def __setstate__(self, data):
-        self.__dict__.update(data)
-
-        if self.binary_stream is None:
-            self.binary_stream = open(self.binary, 'rb')
-        else:
-            self.binary_stream.stream = open(self.binary, 'rb')
-
-        self.reader = elffile.ELFFile(self.binary_stream)
-        if self._dynamic and 'DT_STRTAB' in self._dynamic:
-            self.strtab = next(x for x in self.reader.iter_segments() if x.header.p_type == 'PT_DYNAMIC')._get_stringtable()
-            if 'DT_SYMTAB' in self._dynamic and 'DT_SYMENT' in self._dynamic:
-                fakesymtabheader = {
-                    'sh_offset': AT.from_lva(self._dynamic['DT_SYMTAB'], self).to_rva(),
-                    'sh_entsize': self._dynamic['DT_SYMENT'],
-                    'sh_size': 0, # bogus size: no iteration allowed
-                    'sh_flags': 0,
-                    'sh_addralign': 0,
-                }
-                self.dynsym = elffile.SymbolTableSection(fakesymtabheader, 'symtab_cle', self.reader, self.strtab)
-                self.dynsym.stream = self.memory
-
-                if 'DT_GNU_HASH' in self._dynamic:
-                    self.hashtable = GNUHashTable(
-                            self.dynsym,
-                            self.memory,
-                            AT.from_lva(self._dynamic['DT_GNU_HASH'], self).to_rva(),
-                            self.arch)
-                elif 'DT_HASH' in self._dynamic:
-                    self.hashtable = ELFHashTable(
-                            self.dynsym,
-                            self.memory,
-                            AT.from_lva(self._dynamic['DT_HASH'], self).to_rva(),
-                            self.arch)
-
```

Comment:
Is it not pickleable anymore?

Reply:
It is now pickleable using the default settings :)